backend/src/main/java/com/google/sps/notifs/EmailNotifier.java

```diff
@@ -0,0 +1,174 @@
+    this.toEmail = toEmail;
+  }
+
+  /**
+   * Creates an authorized Credential object.
+   *
+   * @param HTTP_TRANSPORT The network HTTP Transport.
+   * @return An authorized Credential object.
+   * @throws IOException If the credentials.json file cannot be found.
+   */
+  private static Credential getCredentials(final NetHttpTransport HTTP_TRANSPORT)
+          throws IOException {
+    // Load client secrets.
+    InputStream in = new FileInputStream(CREDENTIALS_FILE_PATH);
+    GoogleClientSecrets clientSecrets =
+            GoogleClientSecrets.load(JSON_FACTORY, new InputStreamReader(in));
+
+    // Build flow and trigger user authorization request.
+    GoogleAuthorizationCodeFlow flow =
+            new GoogleAuthorizationCodeFlow.Builder(
+                    HTTP_TRANSPORT, JSON_FACTORY, clientSecrets, SCOPES)
+                    .setDataStoreFactory(
+                            new FileDataStoreFactory(new java.io.File(TOKENS_DIRECTORY_PATH)))
+                    .setAccessType("offline")
+                    .build();
+    LocalServerReceiver receiver = new LocalServerReceiver.Builder().setPort(8000).build();
+    return new AuthorizationCodeInstalledApp(flow, receiver).authorize("user");
+  }
+
+  /**
+   * Create a MimeMessage using the parameters provided.
+   *
+   * @param toEmail email address of the receiver
+   * @param subject subject of the email
+   * @param bodyText body text of the email
+   * @return the MimeMessage to be used to send email
+   * @throws MessagingException for other failures
+   */
+  public static MimeMessage createEmail(String toEmail, String subject, String bodyText)
+          throws MessagingException {
+
+    Properties props = new Properties();
+    Session session = Session.getDefaultInstance(props, null);
+
+    MimeMessage email = new MimeMessage(session);
+
+    email.setFrom(new InternetAddress("Adlib-Step@gmail.com"));
+    email.addRecipient(javax.mail.Message.RecipientType.TO, new InternetAddress(toEmail));
+    email.setSubject(subject);
+    email.setText(bodyText);
+    return email;
+  }
+
+  /**
+   * Create a message from an email.
+   *
+   * @param emailContent Email to be sent to raw of message
+   * @return a message containing a base64url encoded email
+   * @throws IOException if an error occurs writing to the stream
+   * @throws MessagingException for other failures
+   */
+  public static Message createMessageWithEmail(MimeMessage emailContent)
+          throws MessagingException, IOException {
+    ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+    emailContent.writeTo(buffer);
+    byte[] bytes = buffer.toByteArray();
+    String encodedEmail = Base64.encodeBase64URLSafeString(bytes);
+    Message message = new Message();
+    message.setRaw(encodedEmail);
+    return message;
+  }
+
+  /** Getter function that returns the string representing the email recipients name . */
+  public String getRecipientName() {
+    return recipientName;
+  }
```

Comment:
are the getters for recipient name and email ever used or do you foresee a future use case? If not we can remove them

Reply:
I was mainly using them in the body text of the email as well as generally in using the create email function would you suggest directly accessing them the email and name or using getters ?

Reply:
It definitely makes sense to use the getters especially if you're concerned that they will be refactored to be "non-pure" i.e. to do more than just returning a field directly. I could go either way but if we keep the getters, they should be private until we have a use-case to access them from outside the class